hail/python/hail/vds/variant_dataset.py

```diff
@@ -326,6 +356,38 @@ def error(msg):
     def _same(self, other: 'VariantDataset'):
         return self.reference_data._same(other.reference_data) and self.variant_data._same(other.variant_data)
 
+    @staticmethod
+    def _add_len(rd):
+        if 'LEN' in rd.entry:
+            return rd
+        if 'END' not in rd.entry:
+            raise ValueError('Need `END` to compute `LEN` in reference data')
+        return rd.annotate_entries(LEN=rd.END - rd.locus.position + 1)
+
+    @staticmethod
+    def _add_end(rd):
+        if 'END' in rd.entry:
+            return rd
+        if 'LEN' not in rd.entry:
+            raise ValueError('Need `LEN` to compute `END` in reference data')
+        return rd.annotate_entries(END=rd.LEN + rd.locus.position - 1)
+
+    @staticmethod
+    def _add_len_end(rd):
+        if 'END' not in rd.entry and 'LEN' not in rd.entry:
+            raise ValueError('One of `END` or `LEN` must be defined in reference data')
+        rd = VariantDataset._add_len(rd)
+        rd = VariantDataset._add_end(rd)
+        return rd
+
+    @staticmethod
+    def _fix_ref_for_write(rd):
```

Comment:
Feel free to answer "yes!"... is `fix_ref_for_write` going to be well understood to mean "switch over to using LEN and not END"? To my naive eyes, this seems like a very general name for a very specific transformation.

Reply:
Yes! However it's only used in one place now, so I'll just inline it.